src/Microsoft.Azure.WebJobs/StorageAccountAttribute.cs

```diff
@@ -27,14 +27,22 @@ public sealed class StorageAccountAttribute : Attribute
         ///   to any INameResolver registered on the JobHostConfiguration to resolve the actual setting name to use.
         /// - Can be an app setting or connection string name of your choosing.
         /// </param>
-        public StorageAccountAttribute(string account)
+        /// <param name="supported">A flags enum indicating which types of Azure Storage accounts are
+        /// supported by this attribute.</param>
+        public StorageAccountAttribute(string account, StorageAccountTypes supported = StorageAccountTypes.All)
         {
             Account = account;
+            Supported = supported;
         }
 
         /// <summary>
         /// Gets the name of the Azure Storage connection string to use.
         /// </summary>
         public string Account { get; private set; }
+
+        /// <summary>
+        /// Gets the disabled storage account types.
+        /// </summary>
+        public StorageAccountTypes Supported { get; private set; }
```

Comment:
@mathewc Any preference on how to proceed?  Look for existence of FUNCTIONS_EXTENSION_VERSION and disable BlobTrigger on premium accounts?

I think it's either that or surface a method (like the storage account attribute) so that script code can control whether these are enabled, unless if you see another option.  Is there a global configuration for the core sdk?